File: cases/taint/go/return-relay-two-hop-negative/return_relay_two_hop_negative.go
package dataflowbench

func dfb_source() int { // DFB-SOURCE: return-two-hop-negative-input
	return 1
}

func firstRelay(value int) int { // DFB-WITNESS: return-two-hop-negative-first
	return value
}

func secondRelay(value int) int { // DFB-WITNESS: return-two-hop-negative-second
	return firstRelay(value)
}

func dfb_sink(value int) {} // DFB-SINK: return-two-hop-negative-sink

func run() {
	_ = secondRelay(dfb_source())
	dfb_sink(0)
}

File: cases/taint/php/return-relay-two-hop-negative/return_relay_two_hop_negative.php
<?php
function dfb_source(): string { // DFB-SOURCE: return-two-hop-negative-input
    return "tainted";
}

function firstRelay(string $value): string { // DFB-WITNESS: return-two-hop-negative-first
    return $value;
}

function secondRelay(string $value): string { // DFB-WITNESS: return-two-hop-negative-second
    return firstRelay($value);
}

function dfb_sink(string $value): void {} // DFB-SINK: return-two-hop-negative-sink

function run(): void {
    $result = secondRelay(dfb_source());
    dfb_sink("clean");
}
